test.js

```diff
@@ -178,11 +178,39 @@ test('should fail with server error', async t => {
 })
 
 test('should fail with handled server error', async t => {
+  t.plan(2)
   const server = createServer()
   server.on('request', (req, res) => {
     server.emit('error', new Error('foo'))
     res.end()
   })
-  server.on('error', (err) => console.error(err))
-  await t.throwsAsync(axiosist(server).get('/'), { message: 'foo' })
+
+  server.on('error', (err) => {
+    t.is(err.message, 'foo')
+  })
+
+  await axiosist(server).get('/')
+
+  t.is(server.listenerCount('error'), 1)
+})
+
+test('should be listened error to once', async t => {
+  const server = createServer()
+
+  server.on('request', (req, res) => {
+    res.end('ok')
+  })
+
+  server.listen()
+
+  const request = axiosist(server)
+  const promises = []
+
+  for (let i = 0; i < 20; i++) {
+    promises.push(() => request.get('/'))
+  }
+
+  await Promise.all(promises.map(promise => promise()))
+
+  t.is(server.listenerCount('error'), 1)
```

Comment:
This doesn't make sense - only the first request will listen to the server error, others will hang if server emits error event.

Reply:
Consider:

```js
test('should be listened error to once', async t => {
  const request = axiosist((req, res) => {
    res.end('ok')
  })
  const promises = []

  for (let i = 0; i < 20; i++) {
    promises.push(() => request.get('/'))
  }

  await Promise.all(promises.map(promise => promise()))
})
```

This does not seem a typical unit test, if you need concurrent requests to the same server with such number, consider stress tests like autocannon.

Reply:
Maybe you are right. I just want to prove the error listening issue.

Reply:
I am thinking about maybe we can just drop this event listener - serious server errors will be transferred to the socket, which is already handled by axios (one example: https://github.com/axios/axios/blob/v1.7.2/lib/adapters/http.js#L571-L574).

Reply:
Usually, when the server encounters an error, it is handled by a try-catch block or its own listener, so I think this error listener can be removed too.